generators/server/templates/src/main/java/package/security/oauth2/OAuth2AuthenticationService.java.ejs

```diff
@@ -88,9 +90,9 @@ public class OAuth2AuthenticationService {
                 log.debug("successfully authenticated user {}", params.get("username"));
             }
             return ResponseEntity.ok(accessToken);
-        } catch (Exception ex) {
```

Comment:
We can throw the correct `InvalidPasswordException` by catching `HttpClientErrorException` instead of a generic `Exception`.  For other errors (no UAA up or connection refused), the behavior is the same as before - the exception is thrown